processor/deltatocumulativeprocessor/internal/data/add.go

```diff
@@ -76,6 +78,21 @@ func (dp ExpHistogram) Add(in ExpHistogram) ExpHistogram {
 		hi.SetScale(lo.Scale())
 	}
 
+	// Downscale if an expected number of buckets after the merge is too large.
+	from := expo.Scale(dp.Scale())
+	to := max(
+		expo.Limit(maxBuckets, from, dp.Positive(), in.Positive()),
+		expo.Limit(maxBuckets, from, dp.Negative(), in.Negative()),
+	)
```

Comment:
@euroelessar @mx-psi Shouldn't this be `min` instead of `max`? What I am seeing is that if I have no negative data points, then no scaling happens and the number of positive buckets still explodes.

I think the tests missed this because they all use `PosNeg`, i.e., use the same positive and negative data points.

Reply:
@tiit-clarifai can you file an issue for this? It definitely looks suspicious

Reply:
Done, https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/37416.

Reply:
My bad, #37432 fixes it. Thanks for letting me know about it! I've also added tests to catch it in the future.